ark/utils/spatial_lda_utils.py

```diff
@@ -1,63 +1,116 @@
-        raise ValueError("cell_table must be a pd.DataFrame")
-
-    if not all([True for x in base_cols if x in cell_table.columns]):
-        raise ValueError("cell table must contain the following columns:{}".format(base_cols))
+    if not all([x in cell_table.columns for x in BASE_COLS]):
+        raise ValueError(
+            "cell table must contain the following columns:{}".format(
+                BASE_COLS))
 
     # Check markers/clusters
     if all([markers is None, clusters is None]):
         raise ValueError("markers and clusters cannot both be None")
     if markers is not None:
         if isinstance(markers, list) and len(markers) == 0:
             raise ValueError("list of marker names cannot be empty")
-        if not isinstance(markers, list) or not all([isinstance(x, str) for x in markers]):
-            raise ValueError("clusters must be a list of integers")
-        assert all([x in cell_table.columns for x in markers])
+        if not isinstance(markers, list) or not all(
+                [isinstance(x, str) for x in markers]):
+            raise TypeError("markers must be a list of strings")
+        if not all([x in cell_table.columns for x in markers]):
+            raise ValueError("all markers must have a column in cell table")
     if clusters is not None:
         if isinstance(clusters, list) and len(clusters) == 0:
             raise ValueError("list of cluster ids cannot be empty")
-        if not isinstance(clusters, list) or not all([isinstance(x, int) for x in clusters]):
-            raise ValueError("markers must be a list of strings")
+        if not isinstance(clusters, list) or not all(
+                [isinstance(x, int) for x in clusters]):
+            raise TypeError("clusters must be a list of integers")
 
-    # Check fovs
-    if fovs != "all_fovs":
-        if not all([isinstance(x, int) for x in fovs]) or not isinstance(fovs, list) or len(fovs) == 0:
-            raise ValueError("fovs must be 'all_fovs' or a list of integers")
 
-    return None
+def check_featurize_cell_table_args(cell_table, feature_by, radius, cell_index):
+    """Checks the input arguments of the featurize_cell_table() function.
+
+    Args:
+        cell_table (dict):
+            A dictionary whose elements are the correctly formatted
+            pd.DataFrames for each field of view.
+        feature_by (str):
+            One of "cluster", "marker", "avg_marker", or "count".
+        radius (int):
+            Pixel radius corresponding to cellular neighborhood size.
+        cell_index (str):
+            Name of the column in each field of view pd.Dataframe indicating
+            reference cells.
+
+    """
+    # Check valid data types
+    if not isinstance(cell_table, dict):
+        raise TypeError("cell_table should be of type 'dict'")
+    if not isinstance(cell_table[1], pd.DataFrame):
+        raise TypeError("cell_table should contain formatted pd.DataFrames")
+    if not isinstance(feature_by, str):
```

Comment:
Since we already check whether `feature_by` has the exact four values that are required, I don't think this test is necessary. I would say a fair number of these are overkill